include/seastar/util/log.hh

```diff
@@ -211,6 +220,23 @@ public:
         return __builtin_expect(level <= _level.load(std::memory_order_relaxed), false) && !silent;
     }
 
+#if FMT_VERSION >= 80000 && defined(SEASTAR_DEBUG)
+    /// logs to desired level if enabled, otherwise we ignore the log line
+    ///
+    /// \param fmt - {fmt} style format string
+    /// \param args - args to print string
+    ///
+    template <typename... Args>
+    void log(log_level level, fmt::format_string<Args...> fmt, Args&&... args) noexcept {
+        if (is_enabled(level)) {
+            lambda_log_writer writer([&] (internal::log_buf::inserter_iterator it) {
+                return fmt::format_to(it, fmt, std::forward<Args>(args)...);
+
+            });
+            do_log(level, writer);
+        }
+    }
+#else
```

Comment:
Should we keep the alternative with fmt::runtime inputs? This would allow logging with runtime generated formats (a bad idea but can have use cases).

There's a simple variant: `mylogger.info("{}", seastar::value_of([&], { return format(fmt::runtime(create_format_string(), arg)) });`, so probably not worth the effort.

Reply:
i don't really like the idea of having runtime format string. do we have a real-world use case for it?